### PR TITLE
adding specific hook for validation after the upgrade

### PR DESCRIPTION
**Pull request by capolrik**
This new hook should be called after an upgrade to check how example-cnf behaves.

### CODE REVIEW

File: upgrade_validation/hooks/install.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    kind: Event
+    state: absent
+    namespace: "{{ dci_openshift_app_ns|default('example-cnf') }}"
+    wait: yes
+    metadata:
+      name: "{{ event.metadata.name }}"
+    loop: "{{ default_events.resources }}"
+    loop_control:
+      loop_var: event
+    when: default_events|length
```
Comment (tonyskapunk) [inline upgrade_validation/hooks/install.yml:23]:
```suggestion
    when:
      - 'resources' in default_events
      - default_events.resources | length
```

File: upgrade_validation/hooks/install.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+---
+- name: Select event from the default T-REX Profile
+  community.kubernetes.k8s_info:
+    api_version: v1
+    kind: Event
+    namespace: "{{ dci_openshift_app_ns|default('example-cnf') }}"
+    field_selectors:
+      - involvedObject.name = trex-app1
```
Comment (tonyskapunk) [inline upgrade_validation/hooks/install.yml:8]: Not sure if it's needed to be without spaces, i.e. `involvedObject.name=trex-app1`, let's test and confirm

Comment (capolrik) [reply upgrade_validation/hooks/install.yml]: You were right! In fact I already faced a similar case here: https://github.com/dci-labs/trident-config/blob/master/hooks/deploy.yml#L130 and it is working with the extra spaces... weird

File: upgrade_validation/hooks/install.yml (hunk 1 above)
Comment (capolrik) [reply upgrade_validation/hooks/install.yml]: Done

**Review by ramperher — COMMENTED** [review]
@capolrik , don't know if you need a positive review here now to merge everything, but just before that, may you add a README under `upgrade_validation` to explain the purpose of these hooks (and the link with nfv-example-cnf-deploy)? And also add a reference to the default README file of this example-cnf-config repo. Thanks!

File: README.md [hunk 3]
```diff
@@ -51,3 +51,7 @@ $ dci-pipeline cnf:ansible_inventory=/var/lib/dci/inventories/dallas/8nodes/clus
 ```
 TBD
 ```
+
+### Upgrade validation
+When run during an upgrade, an extra DCI hook can be called to gather information from the CNF behavior.
```
Comment (ramperher) [inline README.md:56]: - run = running; or maybe you can directly say "When running an upgrade"
- "from the CNF behavior" -> "about the CNF behavior"?

Comment (capolrik) [reply README.md:56]: You are rigth, about the CNF behavior looks better, but I don't see the pb with the first one.

Comment (ramperher) [reply README.md]: You're right also with my first comment, it's fine as it is right now, I'll close this

**Pull request merged**